language/libra-vm/src/libra_vm.rs

```diff
@@ -332,115 +354,112 @@ impl LibraVM {
         payload: VerifiedTransactionPayload,
         account_currency_symbol: &IdentStr,
     ) -> TransactionOutput {
-        let gas_schedule = match self.get_gas_schedule() {
+        let gas_schedule = match gas_schedule!(self) {
```

Comment:
Why do we need to change the macro over here?

Reply:
Good question. TLDR: to make the borrow checker happy. This all comes down to a limitation that borrow checking stops at function boundaries. When you have a struct with multiple fields, it is perfectly fine to borrow the fields in a disjoint manner as the borrow checker is smart enough to track each field separately. However such information is not available across function boundaries. If you call a function that takes a reference to the struct, the borrow checker has to assume the whole struct is borrowed, even if the function only uses some of the fields.
```
struct Foo {
    counter: usize,
    v: Vec<usize>,
}

fn bar(v: &Vec<usize>) {
    println!("{:?}", v);
}

impl Foo {
    fn f(&mut self) {
        let v = &self.v;
        self.counter += 1; // good
        bar(v);
    }
    
    fn g(&mut self) {
        let v = &self.v;
        self.inc_counter(); // bad
        bar(v);
    }
    
    fn inc_counter(&mut self) {
        self.counter += 1;
    }
}
```

With that being said, I'll take another look at this and see if the macro is necessarily or I can just write the code inlined.